internal/pkg/cli/file/watch_integration_test.go

```diff
@@ -72,60 +73,71 @@ func TestRecursiveWatcher(t *testing.T) {
 
 	t.Run("Watch", func(t *testing.T) {
 		// SETUP
-		err := watcher.Add(fmt.Sprintf("%s/watch", tmp))
+		err := watcher.Add(filepath.ToSlash(filepath.Join(tmp, "watch")))
 		require.NoError(t, err)
 
 		eventsCh := watcher.Events()
 		errorsCh := watcher.Errors()
 
-		expectAndPopulateEvents := func(t *testing.T, n int, events map[fsnotify.Event]struct{}) {
-			for i := 0; i < n; i++ {
+		// expectNextEvents consumes the watcher's event channel to retrieve the next n events in expectedEvents in no particular order.
+		expectNextEvents := func(n int) {
+			for {
+				var e fsnotify.Event
 				select {
-				case e := <-eventsCh:
-					events[e] = struct{}{}
+				case e = <-eventsCh:
 				case <-time.After(time.Second):
+					return
+				}
+
+				if slices.Contains(eventsExpected, e) && !slices.Contains(eventsActual, e) {
+					eventsActual = append(eventsActual, e)
+					n -= 1
+				}
+
+				if n == 0 {
+					return
 				}
 			}
 		}
 
 		// WATCH
-		file, err := os.Create(fmt.Sprintf("%s/watch/subdir/testfile", tmp))
+		file, err := os.Create(filepath.Join(tmp, "watch/subdir/testfile"))
 		require.NoError(t, err)
-		expectAndPopulateEvents(t, 1, eventsActual)
+		expectNextEvents(1)
 
-		err = os.Chmod(fmt.Sprintf("%s/watch/subdir/testfile", tmp), 0755)
+		err = os.Chmod(filepath.Join(tmp, "watch/subdir/testfile"), 0755)
 		require.NoError(t, err)
-		expectAndPopulateEvents(t, 1, eventsActual)
+		expectNextEvents(1)
 
-		err = os.WriteFile(fmt.Sprintf("%s/watch/subdir/testfile", tmp), []byte("write to file"), fs.ModeAppend)
+		err = os.WriteFile(filepath.Join(tmp, "watch/subdir/testfile"), []byte("write to file"), fs.ModeAppend)
 		require.NoError(t, err)
-		expectAndPopulateEvents(t, 2, eventsActual)
+		expectNextEvents(1)
 
 		err = file.Close()
 		require.NoError(t, err)
 
-		err = os.Rename(fmt.Sprintf("%s/watch/subdir", tmp), fmt.Sprintf("%s/watch/subdir2", tmp))
+		err = os.Rename(filepath.Join(tmp, "watch/subdir"), filepath.Join(tmp, "watch/subdir2"))
 		require.NoError(t, err)
-		expectAndPopulateEvents(t, 3, eventsActual)
+		expectNextEvents(2)
```

Comment:
why is the expected even number changed from 3 to 2?

Reply:
Before it would consume duplicate `write` events and so this was accounting for it. I've made it discard duplicate events now.